File: test/fixtures/php/post_image.php
<?php
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'http://localhost:28139/targetservice');
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
curl_setopt($ch, CURLOPT_POSTFIELDS, [
    'image' => new CURLFile('image.jpg'),
]);

$response = curl_exec($ch);

curl_close($ch);

File: test/fixtures/kotlin/post_image.kt
import java.io.File
import java.io.IOException
import okhttp3.MultipartBody
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.asRequestBody

val client = OkHttpClient()

val requestBody = MultipartBody.Builder()
  .setType(MultipartBody.FORM)
  .addFormDataPart("image", "image.jpg", File("image.jpg").asRequestBody())
  .build()

val request = Request.Builder()
  .url("http://localhost:28139/targetservice")
  .post(requestBody)
  .build()

client.newCall(request).execute().use { response ->
  if (!response.isSuccessful) throw IOException("Unexpected code $response")
  response.body!!.string()
}
